File: Foothill College/Python for Programmers/Assignment7/employee.py
class Employee:
   """
   One object of class Employee represents one employee
   """
   def __init__(self, firstName, lastName, socialSecurityNumber, baseSalary):
      '''
      Initializes an Employee object

      firstName and lastName (string): the name of the employee
      socialSecurityNumber (string): ssn, including dashes:
      baseSalary (float): current annual salary

      an Employee object has two attributes:
         self.firstName
         self.lastName
         self.ssn
         self.salary
      '''
      if type(firstName) == str and type(lastName) == str and type(socialSecurityNumber) == str and type(baseSalary) == float:
         self.firstName = firstName
         self.lastName = lastName
         self.ssn = socialSecurityNumber
         self.salary = baseSalary
      else:
         raise TypeError("Invalid employee details")

   def __str__(self):
      return "%s %s is an employee of the firm whose social security number is %s and salary is %d" % (self.firstName, self.lastName, self.ssn, self.salary)


   def giveRaise(self, percentRaise):
      '''
      Increases employee salary by input percentage (float)
      '''
      if (percentRaise > 0):
         self.salary = self.salary * (1 + percentRaise)
      else:
         raise ValueError("Invalid raise percentage")

File: Foothill College/Java Programming/Employee/src/models/Employee.java
/*
* One object of class Employee represents one employee
*
* Author: Hunter S. Carlisle
 */

package models;

public class Employee {
    private String firstName;
    private String lastName;
    private int socialSecurityNumber;
    private double currentSalary;

    // Constructors
    public Employee(){
        // Initialize Employee object with no parameters.
        this.firstName = "";
        this.lastName = "";
        this.socialSecurityNumber = 0;
        this.currentSalary = 0.0;
    }
    public Employee(String firstName, String lastName, int socialSecurityNumber, double baseSalary){
        /*
        Initialize Employee object with parameters, as follows:

        firstName (String): Employee's first name
        lastName (String): Employee's last name
        socialSecurityNumber (int): Employee's government issued SSN
        baseSalary (double): Employee's starting salary upon initialization
         */
        this.firstName = firstName;
        this.lastName = lastName;
        this.socialSecurityNumber = socialSecurityNumber;
        this.currentSalary = baseSalary;
    }

    // getters
    public String getFirstName() { return firstName; }
    public String getLastName() { return lastName; }
    public int getSocialSecurityNumber() { return socialSecurityNumber; }
    public double getSalary() { return currentSalary; }

    // setters
    public void setFirstName(String firstName) { this.firstName = firstName; }
    public void setLastName(String lastName) { this.lastName = lastName; }
    public void setSocialSecurityNumber(int socialSecurityNumber) { this.socialSecurityNumber = socialSecurityNumber; }
    public void setSalary(double baseSalary) { this.currentSalary = baseSalary; }

    // Public Methods
    public String toString(){
        // Print employee object information as a string sentence.
        return (this.firstName + " " + this.lastName + " is an employee of the firm whose social security number is " + this.socialSecurityNumber + " and salary is " + this.currentSalary);
    }

    public void giveRaise(double percentRaise) {
        // EFFECTS: Increase salary by an input
        // REQUIRES: percent raise should be double between 0.0 and 1.0
        currentSalary *= 1.0 + percentRaise;
    }
}
